### ISSUE TITLE
[Bug]: 'NoneType' object has no attribute 'detach' in chatglm sft training 

### ISSUE THREAD

**Issue by wj-Mcat** [PaddlePaddle/PaddleNLP#7058]
### 软件环境

```Markdown
paddle2onnx               1.0.9                                                                                                                                    
paddlefsl                 1.1.0                                                                                                                                    
paddlenlp-ops             0.0.0                                                                                                                                    
paddleocr                 2.6.1.3                                                                                                                                  
paddlepaddle-gpu          0.0.0.post118
```

### 重复问题

- [X] I have searched the existing issues

### 错误描述

```Markdown
如下所示
```

### 稳定复现步骤 & 代码

* 调整训练 json 文件

```json
{
    "model_name_or_path": "__internal_testing__/tiny-fused-chatglm",
    "dataset_name_or_path": "./data",
    "output_dir": "./checkpoints/chatglm_sft_ckpts",
    "per_device_train_batch_size": 4,
    "gradient_accumulation_steps": 4,
    "per_device_eval_batch_size": 8,
    "eval_accumulation_steps":16,
    "num_train_epochs": 3,
    "learning_rate": 3e-05,
    "warmup_steps": 30,
    "logging_steps": 1,
    "max_steps": 10,
    "evaluation_strategy": "epoch",
    "save_strategy": "epoch",
    "src_length": 1024,
    "max_length": 2048,
    "fp16": true,
    "fp16_opt_level": "O2",
    "do_train": true,
    "do_eval": true,
    "disable_tqdm": true,
    "load_best_model_at_end": true,
    "eval_with_do_generation": false,
    "metric_for_best_model": "accuracy",
    "recompute": true,
    "save_total_limit": 1,
    "tensor_parallel_degree": 1,
    "pipeline_parallel_degree": 1
  }
```

* 执行命令

```shell
python finetune_generation.py ./chatglm/sft_argument.json
```

**Comment by lugimzzz** [comment]
chatglm如果要使用trainer需要use_cache设为true，这是chatglm原始组网的问题
